draft-ietf-jsonpath-base.md

```diff
@@ -1671,6 +1639,31 @@ that is the second argument; the result is `LogicalTrue` if such a
 substring exists and `LogicalFalse` otherwise.
 
 
+### `value` Function Extension {#value}
+
+Parameters:
+: 1. `NodesType`
+
+Result:
+: `ValueType`
+
+The "value" function extension provides a way to convert a instance of `NodesType` to a value and
+make that available for further processing in the filter expression:
+
+~~~ JSONPath
+$[?value(@..color) == "red"]
+~~~
+
+Its only argument is an instance of `NodesType` (possibly taken from a
+`filter-path` as in the example above).  The result is an
+instance of `ValueType`.
+
+* If the argument contains a single node, the result is
+  the value of the node.
+* If the argument is `Nothing` or contains multiple nodes, the
+  result is `Nothing`.
```

Comment:
This is a gotcha.  It produces potentially unexpected behavior in expressions.

(Yes, I understand that its behavior is specified, but users won't read the spec.)

This is no different than the implicit conversion that exists in the current `main` document.  The problem is the surprise.  This is why I included the table of behaviors that has been deleted in this PR.

The behavior we define shouldn't contain surprises for users.  That just generates StackOverflow questions.  It should be preferred to reduce the number of questions by devising a system that is intuitive.  This isn't it.  (Arguably, what's in `main` also isn't it.)

Reply:
The point of putting this behaviour inside a function is to make opting in to the behaviour a conscious choice.